File: LeetCode/py/728.py
#Link: https://leetcode.com/problems/self-dividing-numbers/

class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        result = []
        for i in range(left,right+1):
            if self.isSelfDividing(i): result.append(i)
        return result
    def isSelfDividing(self, num : int) -> bool:
        for digit in [int(x) for x in str(num)]:
            if digit == 0 or num%digit != 0: return False
        return True
        

File: LeetCode/cpp/728.cpp
//Link: https://leetcode.com/problems/self-dividing-numbers/

class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> result;
        for(int i = left ; i <= right ; i++)
            if(isSelfDividing(i))
                result.push_back(i);
        return result;
    }
    bool isSelfDividing(int num) {
        for(auto digit : to_string(num)) {
            digit = digit-'0';
            if(digit == 0 || num%digit != 0)
                return false;
        }
        return true;
    }
};
